src/hypothesis/strategies.py

```diff
@@ -542,6 +542,41 @@ def frozensets(elements=None, min_size=None, average_size=None, max_size=None):
 
 
 @defines_strategy
+def iterables(elements=None, min_size=None, average_size=None, max_size=None,
+              unique_by=None, unique=False):
+    """This has the same behaviour as lists, but returns iterables instead.
+
+    It is easy to forget that some iterables cannot be indexed (eg sets)
+    and some do not support indexing (eg generators).  This strategy
+    generates iterators, ensuring that you do not accidentally depend on
+    sequence behaviour.
+
+    """
+    class PrettyIter(object):
+        def __init__(self, values):
+            self._values = values
+            self._iter = iter(self._values)
+
+        def __iter__(self):
+            return self._iter
+
+        if PY2:  # pragma: no cover
+            def next(self):
+                return self._iter.next()
+        else:
+            def __next__(self):
+                return next(self._iter)
+
+        def __repr__(self):
+            return 'iter({})'.format(self._values)
```

Comment:
Actually, one more thing: I'd change this to be `{!r}`, so you get a repr rather than a string representation of `self._values`. Difference between:

```pycon
>>> x = PrettyIter('abcd'); repr(x)
iter(abcd)
```

and

```pycon
>>> x = PrettyIter('abcd'); repr(x)
iter('abcd')
```

It shouldn’t make a difference because we’re only using lists for now, but we might as well do the right thing.